synapse/rest/client/v2_alpha/room_keys.py

```diff
@@ -209,9 +209,15 @@ def on_GET(self, request, room_id, session_id):
         )
 
         if session_id:
-            room_keys = room_keys['rooms'][room_id]['sessions'][session_id]
+            if room_keys['rooms'] == {}:
```

Comment:
I'm finding this a bit opaque, and I think it could do with some comments to explain what's going on.

It feels odd to me that, if the room is missing from the backup,  we 404 if the session_id is passed, but not if it isn't.